presto-main/src/main/java/com/facebook/presto/operator/scalar/StringFunctions.java

```diff
@@ -359,6 +362,49 @@ public static Slice splitPart(@SqlType(StandardTypes.VARCHAR) Slice string, @Sql
         return null;
     }
 
+    @Description("creates a map using delimiter1 and delimiter2")
+    @ScalarFunction
+    @SqlType("map<varchar,varchar>")
+    public static Slice splitToMap(@SqlType(StandardTypes.VARCHAR) Slice string, @SqlType(StandardTypes.VARCHAR) Slice delimiter1, @SqlType(StandardTypes.VARCHAR) Slice delimiter2)
+    {
+        checkCondition(delimiter1.length() > 0, INVALID_FUNCTION_ARGUMENT, "The delimiter1 may not be the empty string");
+        checkCondition(delimiter2.length() > 0, INVALID_FUNCTION_ARGUMENT, "The delimiter2 may not be the empty string");
+
+        Map<Slice, Slice> map = new HashMap<>();
+        int index = 0;
+        while (index < string.length()) {
+            int splitIndex = string.indexOf(delimiter1, index);
+            if (splitIndex < 0) { // Cannot be split by delimiter1
+                break;
+            }
+            // Extract key-value pair based on current index
+            // then add the pair if it can be split by delimiter2
+            Slice kvPair = string.slice(index, splitIndex - index);
+            Slice key    = splitPart(kvPair, delimiter2, 1);
```

Comment:
Use `indexOf` and `slice` as you did for `delimiter1`. `splitPart` does quite a lot of magic under the hood and is designed more toward use directly from SQL.